TheSadRogue.Primitives/Rectangle.cs

```diff
@@ -903,5 +904,86 @@ public IEnumerable<Point> MaxXPositions()
             Direction.Types.Left => MinXPositions(),
             _ => throw new Exception("Cannot retrieve positions on a non-cardinal side of a rectangle.")
         };
+
+        #region division
+        /// <summary>
+        /// Recursively divides a rectangle in half until the small rectangles are between minimumDimension and 2 * minimumDimenson.
+        /// </summary>
+        /// <param name="minimumDimension">The smallest allowable dimension for a rectangle to be.</param>
+        /// <returns>A list of all rectangles that add up to the original rectangle</returns>
+        public IEnumerable<Rectangle> Divide(int minimumDimension)
+        {
+            List<Rectangle> ogChildren = DivideInHalf().ToList();
+            List<Rectangle> children = new List<Rectangle>(); //so that we can modify children during the loop
+            foreach (Rectangle child in ogChildren)
+            {
+                if (child.Width < minimumDimension * 2 && child.Height < minimumDimension * 2)
```

Comment:
Small quibble, but coding convention dictates that we should omit the braces on single-line statements.  This isn't documented anywhere, apologies.